test/library/calibration/test_drag.py

```diff
@@ -120,6 +121,43 @@ def test_nasty_data(self, freq, amp, offset, reps, betas, tol):
         self.assertTrue(abs(result.value.n - backend.experiment_helper.ideal_beta) < tol)
         self.assertEqual(result.quality, "good")
 
+    def test_drag_reanalysis(self):
+        """Test edge case for re-analyzing DRAG result multiple times."""
+        drag_experiment_helper = DragHelper(gate_name="Drag(xp)")
+        backend = MockIQBackend(drag_experiment_helper)
+
+        drag = RoughDrag([1], self.x_plus)
+        drag.set_experiment_options(reps=[2, 4, 6])
+
+        expdata = drag.run(backend, analysis=None)
+        self.assertExperimentDone(expdata)
+
+        # Assume the situation we reloaded experiment data from server
+        # and prepared analysis class in the client machine.
+        # DRAG reps numbers might be different from the default value,
+        # but the client doesn't know the original setting.
+        analysis = DragCalAnalysis()
+        expdata1 = analysis.run(expdata.copy(), replace_results=True)
```

Comment:
Oh, I see what caused the error. `BaseAnalysis.run()` creates an analysis callback so we need to block to make sure it finishes before moving on here.

```suggestion
        expdata1 = analysis.run(expdata.copy(), replace_results=True).block_for_results()
```

Reply:
Or maybe it is more idiomatic to use `self.assertExperimentDone()` on `expdata1` here.

Reply:
Thanks for catching this. I completely forgot to block.